backend/logs_helper/views.py

```diff
@@ -0,0 +1,65 @@
+import json
+import logging
+from datetime import datetime, timezone
+
+from django.conf import settings
+from django.http import HttpRequest
+from rest_framework import status, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from utils.cache_service import CacheService
+from utils.user_session import UserSessionUtils
+
+from .serializers import StoreLogMessagesSerializer
+
+logger = logging.getLogger(__name__)
+
+
+class LogsHelperViewSet(viewsets.ModelViewSet):
+    """Viewset to handle all Tool Studio prompt related API logics."""
+
+    @action(detail=False, methods=["get"])
+    def get_logs(self, request: HttpRequest) -> Response:
+        # Extract the session ID
+        session_id: str = UserSessionUtils.get_session_id(request=request)
+
+        # Construct the Redis key pattern to match keys
+        # associated with the session ID
+        redis_key = f"logs:{session_id}*"
+
+        # Retrieve keys matching the pattern
+        keys = CacheService.get_all_keys(redis_key)
+
+        # Retrieve values corresponding to the keys and sort them by timestamp
+        logs = []
+        for key in keys:
+            log_data = CacheService.get_key(key)
+            if log_data:
+                log_entry = json.loads(log_data)
+                logs.append(log_entry)
+
+        # Sort logs based on timestamp
+        sorted_logs = sorted(logs, key=lambda x: x["timestamp"])
+
+        return Response({"data": sorted_logs}, status=status.HTTP_200_OK)
+
+    @action(detail=False, methods=["post"])
+    def store_log(self, request: HttpRequest) -> Response:
```

Comment:
@jagadeeswaran-zipstack why do we have this API? When / how will FE be calling this?

Reply:
We're storing notifications from the frontend. For instance, when an adapter is created, we'll get a notification that says 'Added Successfully'. These messages are stored using this API.

Reply:
Does this mean that the notifications we raise from the backend to the frontend are once again sent to the backend for persistence?

Reply:
Yes, you're correct. However, only a few logs from the backend will be stored in Redis from backend itself. We are doing this to avoid missing any logs. In certain cases, such as when we add an adapter, I don't believe we store those logs, thus this will be handled by FE.